src/ResourceManager/Websites/Commands.Websites/Strategies/PSCmdletAdapter.cs

```diff
@@ -0,0 +1,241 @@
+
+        ConcurrentQueue<ShouldProcessPrompt> _process = new ConcurrentQueue<ShouldProcessPrompt>();
+
+        ConcurrentQueue<ShouldContinuePrompt> _continue = new ConcurrentQueue<ShouldContinuePrompt>();
+
+        ConcurrentQueue<ProgressRecord> _progress = new ConcurrentQueue<ProgressRecord>();
+
+        ConcurrentDictionary<string, double> _activity = new ConcurrentDictionary<string, double>();
+
+        object _lock = new object();
+
+        int _hasMessages = 0;
+
+        protected int Retries { get; set; }
+
+        public TimeSpan RetryInterval { get; set; }
+
+        public SyncTaskScheduler Scheduler { get; } = new SyncTaskScheduler();
+
+        public Task<bool> ShouldChangeAsync(string target, string action)
+        {
+            var process = new ShouldProcessPrompt { Target = target, Message = action, Completer = new TaskCompletionSource<bool>() };
+            _process.Enqueue(process);
+            _hasMessages = 1;
+            return process.Completer.Task;
+        }
+
+        public Task<bool> ShouldContinueChangeAsync(string query, string caption)
+        {
+            var process = new ShouldContinuePrompt { Query = query, Caption = caption, Completer = new TaskCompletionSource<bool>() };
+            _continue.Enqueue(process);
+            _hasMessages = 1;
+            return process.Completer.Task;
+        }
+
+        public void WriteExceptionAsync(Exception exception)
+        {
+            var error = new ErrorRecord(exception, "ExecutionException", ErrorCategory.InvalidOperation, exception.TargetSite);
+            _error.Enqueue(error);
+            _hasMessages = 1;
+        }
+
+        public void WriteVerboseAsync(string verboseMessage)
+        {
+            _verbose.Enqueue(verboseMessage);
+            _hasMessages = 1;
+        }
+
+        public void WriteDebugAsync(string debugMessage)
+        {
+            _debug.Enqueue(debugMessage);
+            _hasMessages = 1;
+        }
+
+        public void WriteWarningAsync(string warningMessage)
+        {
+            _warning.Enqueue(warningMessage);
+            _hasMessages = 1;
+        }
+
+        void WriteProgressAsync(ProgressRecord progress)
+        {
+            _progress.Enqueue(progress);
+            _hasMessages = 1;
+        }
+
+        public void ReportTaskProgress(ITaskProgress taskProgress)
+        {
+            var progress = 0.0;
+            var config = taskProgress.Config;
+            string key = config.Name + " " + config.Strategy.Type;
+            if (_activity.ContainsKey(key))
+            {
+                progress = _activity[key];
+            }
+
+            progress += taskProgress.GetProgress();
+            _activity[key] = progress;
+            var percent = (int)(progress * 100.0);
+            var r = new[] { "|", "/", "-", "\\" };
+            var x = r[DateTime.Now.Second % 4];
+            WriteProgressAsync(
```

Comment:
this implementation may have a bug in progress bar. Here's a fix for it https://github.com/Azure/azure-powershell/pull/5487/files